src/System.Data.SqlClient/src/System/Data/SqlClient/LocalDBAPI.Windows.cs

```diff
@@ -0,0 +1,48 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+
+using System.Threading;
+using System.Data.SqlClient;
+
+namespace System.Data
+{
+    internal static partial class LocalDBAPI
+    {
+
+        private static IntPtr UserInstanceDLLHandle
+        {
+            get
+            {
+                if (s_userInstanceDLLHandle == IntPtr.Zero)
+                {
+                    bool lockTaken = false;
+                    try
+                    {
+                        Monitor.Enter(s_dllLock, ref lockTaken);
```

Comment:
The code is using:
```C#
bool lockTaken = false;
try
{
    Monitor.Enter(s_dllLock, ref lockTaken);
    ... // protected code
}
finally
{
    if (lockTaken) Monitor.Exit(s_dllLock);
}
```
That's exactly what ```lock``` does, so it could just be:
```C#
lock (s_dllLock)
{
    ... // protected code
}
```